src/app/auth/auth.component.ts

```diff
@@ -1,47 +1,69 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, OnInit, OnDestroy } from '@angular/core';
 import { ActivatedRoute, Router } from '@angular/router';
 import {
     FormBuilder,
     FormGroup,
     FormControl,
     Validators,
 } from '@angular/forms';
+import { Subscription } from 'rxjs';
+import { AuthModel } from '../models/auth.model';
+import { AuthService } from '../services';
 
 @Component({
     selector: 'app-login',
     templateUrl: './auth.component.html',
     styleUrls: ['./auth.component.less'],
 })
-export class AuthComponent implements OnInit {
+export class AuthComponent implements OnInit, OnDestroy {
     authType: String = '';
```

Comment:
```suggestion
    authType: string = '';
```

the same for other places.